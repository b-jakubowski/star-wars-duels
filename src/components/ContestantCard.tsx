import React from 'react';
import { Card, CardContent, CardHeader, makeStyles, Typography } from '@material-ui/core';

interface ContestantCardProps {
	name: string;
	unit: string;
	image: string;
}

const useStyles = makeStyles({
	link: {
		textDecoration: 'none',
	},
	card: {
		marginTop: 36,
	},
	content: {
		display: 'flex',
		alignItems: 'center',
		flexDirection: 'column',
		width: '100%',
	},
	img: {
		height: 220,
		width: 'auto',
	},
	title: {
		textAlign: 'center',
	},
});

const ContestantCard: React.FC<ContestantCardProps> = ({ name, unit, image }: ContestantCardProps) => {
	const classes = useStyles();

	return (
		<Card className={classes.card}>
			<CardContent className={classes.content}>
				<CardHeader data-testid="contestant-card-title" className={classes.title} title={name} />
				<img className={classes.img} src={image} alt={image} />
				<Typography data-testid="contestant-card-unit" color="textSecondary">
					{unit}
				</Typography>
			</CardContent>
		</Card>
	);
};

export default ContestantCard;
