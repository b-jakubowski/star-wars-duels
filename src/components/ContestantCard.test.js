import React from 'react';
import { render } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import ContestantCard from './ContestantCard';
import starship1 from '../images/millennium-falcon-011.jpg';

test('ContestantCard receives props, renders title text and unit', () => {
	const { getByTestId } = render(
		<Router>
			<ContestantCard name={'Death Star'} unit={`mass: 73228`} image={starship1} />
		</Router>,
	);

	expect(getByTestId('contestant-card-title')).toHaveTextContent('Death Star');
	expect(getByTestId('contestant-card-unit')).toHaveTextContent('mass: 73228');
});

it('ContestantCard renders correctly', () => {
	const tree = renderer.create(<ContestantCard name={'Death Star'} unit={`mass: 73228`} image={starship1} />).toJSON();
	expect(tree).toMatchSnapshot();
});
