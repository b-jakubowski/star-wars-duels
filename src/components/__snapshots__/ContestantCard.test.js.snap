// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContestantCard renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-73 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root makeStyles-content-74"
  >
    <div
      className="MuiCardHeader-root makeStyles-title-76"
      data-testid="contestant-card-title"
    >
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          Death Star
        </span>
      </div>
    </div>
    <img
      alt="millennium-falcon-011.jpg"
      className="makeStyles-img-75"
      src="millennium-falcon-011.jpg"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      data-testid="contestant-card-unit"
    >
      mass: 73228
    </p>
  </div>
</div>
`;
